src/EllCrv/NewtonLift.jl

```diff
@@ -0,0 +1,302 @@
+#import Hecke.AbstractAlgebra
+export newton_lift
+add_verbosity_scope(:NewtonIteration)
+
+_valuation(x::RingElem) = iszero(x) ? QQ(precision(parent(x))) : valuation(x)
+
+
+r"""
+    _singular_points_weierstrass_model(E::EllipticCurve{<:FracField{<:AbstractAlgebra.Generic.Poly}})
+
+Return the singular points of the weierstrass model W/k of E/k(t).
+
+It is assumed that they are all defined over k.
+"""
+function _singular_points_weierstrass_model(E::EllipticCurve{<:AbstractAlgebra.Generic.FracFieldElem{<:AbstractAlgebra.Generic.Poly}})
+  p = [i[1] for i in factor(numerator(discriminant(E))) if i[2]>1]
+  Ft = base_ring(base_field(E))
+  F = coefficient_ring(Ft)
+  P,x = polynomial_ring(F,:x)
+  a1, a2, a3, a4, a6 = a_invariants(E)
+
+  #f = y^2 + a1*x*y + a3*y - (x^3 + a2*x^2 + a4*x + a6)
+  @req iszero(a1) "not implemented for general weierstrass models"
+  @req iszero(a3) "not implemented for general weierstrass models"
+  sing = Vector{elem_type(F)}[]
+  for cc in p
+    @req degree(cc)==1 "all reducible singular fibers need to be defined over the base field"
+    t0 = -constant_coefficient(cc)
+    a1c, a2c, a3c, a4c, a6c = [evaluate(a,t0) for a in a_invariants(E)]
+    f = (x^3 + a2c*x^2 + a4c*x + a6c)
+    pt = [i[1] for i in factor(f) if i[2]>1]
+    @assert length(pt)==1
+
+    pt = pt[1]
+    x0 = -constant_coefficient(pt)
+    y0 = F(0)
+    push!(sing, [x0,y0,t0])
+  end
+  return sing
+end
+
+@doc r"""
+    newton_lift(E::EllipticCurve, P::EllipticCurvePoint, reduction_map; max_iterations=16) -> Bool, EllipticCurvePoint
```

Comment:
The OSCAR tests fail because of this doc string.
```suggestion
@doc raw"""
    newton_lift(E::EllipticCurve, P::EllipticCurvePoint, reduction_map; max_iterations=16) -> Bool, EllipticCurvePoint
```
seems to be better.

Reply:
thanks for spotting this! `r` was the python way.